### DISCUSSION TITLE
How to Get Android Wifi List on Maui

### DISCUSSION THREAD

**Discussion by HaonPAPA** [dotnet/maui#9339] [General]
My first encounter with Maui. I'm going to do some tests. Is there any way to get the wifi list??

Is Maui based on xamarin?
Can I use all of the Xamarin libraries?

**Comment by PirasathLuxchumykanthan** [comment]
no u cant.

**Reply by HaonPAPA** [reply]
Is there any way I can get the wifi list?

**Reply by VWilcox2000** [reply]
For iOS, you can join the MFi program to connect to your devices, but I am not sure if even then you can see the wifi list.  Android went from the best option to the worst -- we already had consenting customers using it very conveniently who similarly did not appreciate them just destroying that whole system rather than putting it behind consent and permissions common sense would dictate.  What we did to get around the inability to list wifi is to use bluetooth advertising to detect what devices are nearby.  We can correlate the bluetooth ad id's to the wifi items.  Android's initial change was to prompt one of the ugliest wifi selection systems I've ever seen that would put our device with the highest signal strength on page 4 in small text -- an embarrassing form factor.  In more recent versions, they've allowed specifying the preferred station and added flags for whether it mattered if it had internet -- but I've seen the system not actually change behavior based on those.  I am hoping Google announces a GFi system similar to MFi, but haven't found anything like it to date.  The good news is, with Android, you can use full up Bluetooth instead of BLE and there are libraries that allow that to work with sockets just like wifi.  So, MFi program (not that difficult to get into) on iOS and Bluetooth on Android overcome most of their lazy hackery, but for listing wifi, you will need to use the bluetooth trick above or have a separate device installed at your site that can query the wifi names in range and inform you.  All that when a simple "ControlYourStuff App is wanting permission to connect to 'xyz' network?  Allow it now, while using app, or any time?" and "ControlYourStuff App is wanting to browse your wifi list for devices it supports?  Allow..."  was 1000% called for on BOTH platforms.  Google is always gonna Google, though, I never got my final version of Google Glass I was promised either... and ask Stadia users how those promises worked out.
